tests/unit/utils/test_boto3mod.py

```diff
@@ -0,0 +1,163 @@
+"""
+    Tests for salt.utils.boto3mod
+"""
+
+import random
+import string
+
+import salt.loader
+import salt.utils.boto3mod as boto3mod
+from salt.utils.versions import LooseVersion
+from tests.support.mixins import LoaderModuleMockMixin
+from tests.support.mock import MagicMock, patch
+from tests.support.unit import TestCase, skipIf
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+REQUIRED_BOTO3_VERSION = "1.2.1"
+
+
+def __virtual__():
+    """
+    Returns True/False boolean depending on if Boto3 is installed and correct
+    version.
+    """
+    if not HAS_BOTO3:
+        return False
+    if LooseVersion(boto3.__version__) < LooseVersion(REQUIRED_BOTO3_VERSION):
+        return (
+            False,
+            (
+                "The boto3 module must be greater or equal to version {}"
+                "".format(REQUIRED_BOTO3_VERSION)
+            ),
+        )
+    return True
+
+
+@skipIf(HAS_BOTO3 is False, "The boto module must be installed.")
+@skipIf(
+    LooseVersion(boto3.__version__) < LooseVersion(REQUIRED_BOTO3_VERSION),
+    "The boto3 module must be greater or equal to version {}".format(
+        REQUIRED_BOTO3_VERSION
+    ),
+)
+class Boto3modTestCase(TestCase, LoaderModuleMockMixin):
+    """
+    TestCase for salt.utils.boto3mod module
+    """
+
+    region = "us-east-1"
+    service = "test-service"
+    resource_name = "test-resource"
+    resource_id = "test-resource-id"
+    access_key = "GKTADJGHEIQSXMKKRBJ08H"
+    secret_key = "askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs"
+    conn_parameters = {}
+    error_message = (
+        "An error occurred ({}) when calling the {} operation: Test-defined error"
+    )
+    error_content = {"Error": {"Code": 101, "Message": "Test-defined error"}}
+    session_ret = {}
+    conn = None
+
+    def setup_loader_modules(self):
+        self.opts = {
+            "__salt__": {"config.option": salt.config.DEFAULT_MINION_OPTS.copy()}
+        }
+        return {boto3mod: self.opts}
+
+    def setUp(self):
+        super().setUp()
+        boto3mod.__virtual__()
```

Comment:
Apologies if I missed this the first time, but do we need this call to `__virtual__()` or event the virtual function at all, since you are skipping the test in the `skipIf` on the test class?

Reply:
That is a great question.  I admit borrowed the pattern from test_botomod, but I agree, it shouldn't be necessary.